code/modules/mob/mob_helpers.dm

```diff
@@ -35,6 +35,19 @@
 /mob/living/carbon/human/isMonkey()
 	return istype(species, /datum/species/monkey)
 
+
+/**
+ * Checks if the target has a grab from the user
+ */
+/mob/proc/has_grab(mob/user)
```

Comment:
`has_upgraded_grab` would be more indicative I think.

Reply:
has_grab(user, grab_level)?

Reply:
That might be a better approach yeah, some flag-based system.

Reply:
Updated the proc name to `has_danger_grab()` since the chain specifically checks the danger var.